tianshou/policy/base.py

```diff
@@ -213,4 +214,11 @@ def compute_nstep_return(
         returns = to_torch_as(returns, target_q)
         gammas = to_torch_as(gamma ** gammas, target_q)
         batch.returns = target_q * gammas + returns
+        # prio buffer update
+        if isinstance(buffer, PrioritizedReplayBuffer):
+            batch.update_weight = buffer.update_weight
```

Comment:
Mounting a function ``buffer.update_weight`` to be a field of batch objects is a hack and should be avoided. Since it has been here for a while and this PR is already very large, I will open a new PR to deal with it.

The solution may be something like the following: add a function ``BasePolicy .update`` and a function ``BasePolicy.post_process_fn``. The update of weight into the buffer can be done in ``BasePolicy.post_process_fn``. Trainer functions just have to call ``BasePolicy.update``.

```python
def update(self, buffer, batch_size):
    batch, indices = buffer.sample(batch_size)
    self.process_fn(batch, buffer, indices)
    self.learn(batch)
    self.post_process_fn(batch, buffer, indices)
```